### PR TITLE
ADEN-2299 Fix spotlights tests

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java [hunk 1]
```diff
@@ -90,7 +90,6 @@ public Object apply(WebDriver driver) {
         }
       });
     } catch (TimeoutException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 4 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/a427f54de2164283451f77fafaaf5dff01251caa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L705) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/a427f54de2164283451f77fafaaf5dff01251caa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L613) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MINOR [Define a constant instead of duplicating this literal "verifyColorAd" 3 times.](https://github.com/Wikia/selenium-tests/blob/a427f54de2164283451f77fafaaf5dff01251caa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L123) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/a427f54de2164283451f77fafaaf5dff01251caa/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L67) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/common/core/imageutilities/Shooter.java [hunk 2]
```diff
@@ -44,8 +45,15 @@ public File capturePage(WebDriver driver) {
    * @return File path  - file's handler which was saved in given path
    */
   public File captureWebElement(WebElement element, WebDriver driver) {
-    Object[] rect = getBoundingClientRect(element, driver);
-    File image = imageEditor.cropImage((Point) rect[0], (Dimension) rect[1], capturePage(driver));
+    Point start = element.getLocation();
+    Dimension size = element.getSize();
+    if (!"FF".equals(Configuration.getBrowser())) {
```
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/common/core/imageutilities/Shooter.java]: So... basically it worked only in FF? :smiling_imp: And probably in Google Chrome when `devicePixelRatio === 1` :smile_cat: If it works now on both browsers, let's remove Notes from method description.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java (hunk 1 above)
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java]: To resolve this Sonar issue you can now re-use @drets's new method: https://github.com/Wikia/selenium-tests/pull/797/files#diff-09e6725174ba92baabf453454b8a1d4aR117 :)

**Comment by nandy-andy** [comment]
Cool! :) Let's just address Sonar's issues and we can move on.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Define a constant instead of duplicating this literal "verifyColorAd" 3 times.](https://github.com/Wikia/selenium-tests/blob/f98edb425b14b37cf8790de5a9d3f5f6443e0a76/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L124) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "ScreenshotsComparison" 3 times.](https://github.com/Wikia/selenium-tests/blob/f98edb425b14b37cf8790de5a9d3f5f6443e0a76/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/fd8ed090aca0c520ac29f7ab8a1846c8a50ede7a/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L67) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 23 issues:
* MAJOR 16 major
* MINOR 7 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L404) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* MINOR [Define a constant instead of duplicating this literal "<tr class=\"" 4 times.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L286) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [Remove this unused "dateFormat" local variable.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L369) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L298) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MINOR [Define a constant instead of duplicating this literal "success" 3 times.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L362) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)
* MAJOR [Introduce a new variable instead of reusing the parameter "imageAsBase64".](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L151) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)
* MAJOR [Either remove or fill this block of code.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L240) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00108)
* MINOR [Define a constant instead of duplicating this literal "error" 3 times.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L295) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MINOR [Define a constant instead of duplicating this literal "</td><td> <br/> &nbsp;</td></tr>" 6 times.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L113) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L144) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L282) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [Remove this unused "date" local variable.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L370) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MAJOR [Remove those useless parentheses.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L153) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)
* MINOR [Define a constant instead of duplicating this literal "</td><td>" 4 times.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L288) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L403) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L321) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L301) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* MINOR [Define a constant instead of duplicating this literal "\"><td>" 4 times.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "ScreenshotsComparison" 3 times.](https://github.com/Wikia/selenium-tests/blob/6eee50588cd39c8f53edc66240c2671cc48a7f15/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)

File: src/test/java/com/wikia/webdriver/common/core/imageutilities/Shooter.java (hunk 2 above)
Comment (drets) [inline src/test/java/com/wikia/webdriver/common/core/imageutilities/Shooter.java]: @fraszczakszymon good catch :+1:

**Comment by drets** [comment]
LGTM

**Pull request merged**